community/modules/scripts/spack-install/templates/install_spack.tpl

```diff
@@ -99,6 +99,7 @@ echo "$PREFIX Installing root spack specs..."
 echo "$PREFIX Configuring spack environments"
 %{if ENVIRONMENTS != null ~}
 %{for e in ENVIRONMENTS ~}
+if [ ! -d ${INSTALL_DIR}/var/spack/environments/${e.name} ]; then
```

Comment:
The only issue I could see with this is if spack ever lets you change the path to where environments live.

It might be better to do something like:
`if [ ! -z "$(spack env list | grep ${e.name}) ]; then`

So you're using spacks internals rather than checking a hardcoded path.

FWIW, I don't think it's currently possible to change the path that named environments are created in though.